internal/external-dns/plan/plan.go

```diff
@@ -200,6 +206,29 @@ func (p *Plan) Error() error {
 	return errors.Join(p.Errors...)
 }
 
+func (p *Plan) processStopLabel() {
+	if p.RootHost == nil {
+		return
+	}
+	desiredTree := common.MakeTreeFromEndpoints(*p.RootHost, p.Desired)
+	currentTree := common.MakeTreeFromEndpoints(*p.RootHost, p.Current)
+
+	// bad trees, return
+	if len(desiredTree.Children) == 0 || len(currentTree.Children) == 0 {
+		return
+	}
+
+	// copy soft_delete and stop_soft_delete labels from desired to current tree for propagation
+	common.CopyLabel(SoftDeleteLabel, desiredTree, currentTree)
+	common.CopyLabel(StopSoftDeleteLabel, desiredTree, currentTree)
+
+	// propagate label in desired tree ignoring stop labels
+	common.PropagateLabel(desiredTree, SoftDeleteLabel, "true")
+
+	// propagate the stop label in the current tree
+	common.PropagateStoppableLabel(currentTree, SoftDeleteLabel, "true", StopSoftDeleteLabel)
+}
+
```

Comment:
I have read through this several times and i don't really understand the flow here. Can you explain it?

* Why do we create two trees instead of a single tree that is a combination of both current and desired (current overlaid with desired) and represents the final combined structure?
* Why do we copy labels from desired to current?
* Why are the `SoftDeleteLabel` labels propagated on desired, but the `SoftDeleteLabel` and `StopSoftDeleteLabel` dealt with on current? Why is current being manipulated at all?